sdk/python/lib/pulumi/runtime/resource.py

```diff
@@ -1113,17 +1113,17 @@ async def _add_dependency(
     * Comp3 and Cust5 because Comp3 is a child of a remote component resource
     """
 
+    # Exit early if there are cycles to avoid hangs.
+    cycles = not declare_dependency(from_resource, res) if from_resource else False
+    if cycles:
+        return
```

Comment:
I think the earlier version of this ternary condition was slightly more obvious to read.
Any opposition to changing to that?

```suggestion
    # Exit early if there are cycles to avoid hangs.
    no_cycles = declare_dependency(from_resource, res) if from_resource else True
    if not no_cycles:
        return
```

I read that as "check(x) if x else assume_good", versus "not check(x) if x else assume_bad".